cpp/benchmarks/io/cuio_common.cpp

```diff
@@ -37,15 +37,25 @@ std::string random_file_in_dir(std::string const& dir_path)
 }
 
 cuio_source_sink_pair::cuio_source_sink_pair(io_type type)
-  : type{type}, file_name{random_file_in_dir(tmpdir.path())}
+  : type{type},
+    d_buffer{0, cudf::get_default_stream()},
+    file_name{random_file_in_dir(tmpdir.path())}
 {
 }
 
 cudf::io::source_info cuio_source_sink_pair::make_source_info()
 {
   switch (type) {
     case io_type::FILEPATH: return cudf::io::source_info(file_name);
-    case io_type::HOST_BUFFER: return cudf::io::source_info(buffer.data(), buffer.size());
+    case io_type::HOST_BUFFER: return cudf::io::source_info(h_buffer.data(), h_buffer.size());
+    case io_type::DEVICE_BUFFER: {
+      auto const stream = cudf::get_default_stream();
```

Comment:
Why are we going to use the default stream? Is there any input stream stored?

Reply:
Because we're returning `source_info` constructed from `d_buffer`. If the default stream is used to copy data to `d_buffer` but some other stream is used to access it then ......

Reply:
This is a case that will need to be fixed in order to make libcudf a fully stream friendly library. However, it is fine for now to make this use the  cudf default stream.